### PR TITLE
test(ci): add flush investigation notebook and CI workflow

### LINKED ISSUE #230: feat(ci): add flush investigation notebook and CI workflow
Add the flush+reset investigation notebook and a GitHub Actions workflow that runs it inside Docker, converts to HTML with embedded audio, and uploads as an artifact with provenance metadata. Depends on Docker build infra from #228.

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary

- Add `notebooks/flush-investigation.ipynb` — systematically compares audio across four flush strategies using MFCC and multi-resolution spectral loss
- Add `.github/workflows/flush-investigation.yml` — runs notebook weekly in Docker with headless X11, uploads HTML artifact with provenance
- Notebook uses hardcoded params (seed=42) so results are reproducible across branches without importing param specs

Depends on #229 (Docker build infra).

Closes #230

## Verification

- [ ] Notebook runs locally with Surge XT plugin present
- [ ] Workflow YAML passes validation
- [ ] Notebook JSON is valid and cells import from `src.data.vst.core`
- [ ] Workflow wraps notebook execution in `run-linux-vst-headless.sh`
- [ ] Provenance metadata appended to HTML artifact

### CODE REVIEW

File: notebooks/flush-investigation.ipynb [hunk 1]
```diff
@@ -0,0 +1,1084 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "STRATEGY_DESC = {\n",
+    "    \"none\": \"post-load flush only (no post-param, no post-render)\",\n",
+    "    \"post_param\": \"post-load + post-param flush (no post-render)\",\n",
+    "    \"post_render\": \"post-load + post-render flush (no post-param)\",\n",
+    "    \"all\": \"post-load + post-param + post-render flush (all three)\",\n",
+    "}\n",
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply notebooks/flush-investigation.ipynb]: Fixed in 4e046e2: STRATEGY_DESC defined once in the helpers cell, removed duplicate.

File: .github/workflows/flush-investigation.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+      - name: Build Docker image (dev-live target)
+        env:
+          GIT_PAT: ${{ secrets.GIT_PAT }}
+          DOCKER_BUILDKIT: "1"
+        run: |
+          make docker-build-dev-live \
+            GIT_REF=${{ github.sha }} \
+            GIT_PAT=${{ secrets.GIT_PAT }} \
+            DOCKER_BUILD_FLAGS="--load"
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply .github/workflows/flush-investigation.yml:46]: Fixed in 4e046e2: added a `Capture checked-out SHA` step using `git rev-parse HEAD`, used consistently for Docker build, provenance, and artifact name.

File: .github/workflows/flush-investigation.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            -c "
+              pip install -q jupyter nbconvert auraloss librosa &&
+              cd /workspace &&
+              scripts/run-linux-vst-headless.sh \
+                jupyter nbconvert \
+                  --to html \
+                  --execute \
+                  --ExecutePreprocessor.timeout=600 \
+                  --output-dir=/reports \
+                  notebooks/flush-investigation.ipynb
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply .github/workflows/flush-investigation.yml]: Fixed in 4e046e2: now mounts only `notebooks/` dir instead of full workspace. Plugin path comes from `SURGE_VST3_PATH` env var (defaults to `/usr/lib/vst3/Surge XT.vst3` in CI).

File: .github/workflows/flush-investigation.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            triggered_by: ${{ github.actor }}
+            timestamp: $(date -u +%Y-%m-%dT%H:%M:%SZ)
+          -->
+          PROVENANCE
+
+      - name: Upload notebook HTML artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: flush-investigation-${{ github.sha }}
+          path: reports/flush-investigation.html
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply .github/workflows/flush-investigation.yml:89]: Fixed in 4e046e2: provenance and artifact name both use `steps.meta.outputs.sha` from `git rev-parse HEAD`.

File: notebooks/flush-investigation.ipynb [hunk 5]
```diff
@@ -0,0 +1,1084 @@
...
+    "import auraloss\n",
+    "import librosa\n",
+    "import torch\n",
+    "from loguru import logger\n",
+    "\n",
+    "logger.disable(\"src.data.vst.core\")\n",
+    "\n",
+    "\n",
+    "def fresh_plugin():\n",
+    "    return VST3Plugin(PLUGIN_PATH)\n",
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply notebooks/flush-investigation.ipynb]: Fixed in 4e046e2: `fresh_plugin()` now uses `load_plugin(PLUGIN_PATH)` which includes the `_prepare_plugin(show_editor)` initialization, matching production semantics.

File: notebooks/flush-investigation.ipynb [hunk 6]
```diff
@@ -0,0 +1,1084 @@
...
+    "from loguru import logger\n",
+    "from pedalboard import VST3Plugin\n",
+    "\n",
+    "from src.data.vst.core import load_preset, make_midi_events, set_params  # noqa: E402\n",
+    "\n",
+    "root = rootutils.setup_root(os.getcwd(), indicator=\".project-root\", pythonpath=True)\n",
+    "os.chdir(root)\n",
+    "logger.disable(\"src.data.vst.core\")\n",
+    "\n",
+    "PLUGIN_PATH = str((root / \"plugins\" / \"Surge XT.vst3\").resolve())\n",
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply notebooks/flush-investigation.ipynb]: Fixed in 4e046e2: notebook reads `SURGE_VST3_PATH` env var first, falls back to repo-local `plugins/`, raises `RuntimeError` with clear message if neither exists.

File: notebooks/flush-investigation.ipynb [hunk 7]
```diff
@@ -0,0 +1,1084 @@
...
+    "PRESET_PATH = \"presets/surge-base.vstpreset\"\n",
+    "SAMPLE_RATE = 44100.0\n",
+    "CHANNELS = 2\n",
+    "DURATION = 2.0\n",
+    "N_RENDERS = 5\n",
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply notebooks/flush-investigation.ipynb]: Fixed in 4e046e2: markdown now clarifies configs are hardcoded constants pre-generated with `seed=42`, no RNG runs at notebook execution time.

File: notebooks/flush-investigation.ipynb [hunk 8]
```diff
@@ -0,0 +1,1084 @@
...
+    "produce different outputs — the model sees noise rather than a clean function. If\n",
+    "renders are not deterministic, we need to understand the magnitude: is it floating-point\n",
+    "jitter (1e-7) the model will ignore, or perceptual-level variation (0.01+) that\n",
+    "effectively adds a random augmentation the model must learn through?\n",
+    "\n",
+    "**Method:** Render  5 times per strategy, each with a fresh plugin instance.\n",
+    "Compare render_0 to all others (sequential pairs), and also compare even-to-even /\n",
+    "odd-to-odd pairs (alternate pairs) to detect binary toggle patterns.\n",
+    "\n",
+    "**How to read results:** If all sequential pairs show , the strategy\n",
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply notebooks/flush-investigation.ipynb]: Fixed in 4e046e2: sentence now reads "If all sequential pairs show near-zero diffs, the strategy is fully deterministic."
